### PR TITLE
ddl: move syncer to a single package

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Problem Summary:

Move syncer to package `syncer` so that it can depend on some packages which rely on ddl/util.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tangenta — APPROVED** [review]

**Review by CbcWestwolf — APPROVED** [review]

File: DEPS.bzl [hunk 1]
```diff
@@ -1568,6 +1568,13 @@ def go_deps():
         sum = "h1:ThE+hJP0fEp4zWLkWHWcRyI2Od0p7DlgYG3Uqrmrcpk=",
         version = "v0.1.1",
     )
+    go_repository(
```
Comment (hawkingrei) [inline DEPS.bzl:1571]: Why have you dependency to sync ```bazel``` without changing ```gomod```?

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**